lib/Target/LLVMIR/ConvertToROCDLIR.cpp

```diff
@@ -0,0 +1,117 @@
+//===- ConvertToROCDLIR.cpp - MLIR to LLVM IR conversion -------------------===//
+//
+// Copyright 2019 The MLIR Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// =============================================================================
+//
+// This file implements a translation between the MLIR LLVM + ROCDL dialects and
+// LLVM IR with ROCDL intrinsics and metadata.
```

Comment:
This seems like another component that I rather review in a separate followup PR from the introduction of the dialect itself.

Reply:
ok will split this PR into smaller pieces and file this module as a separate PR.